collect_app/src/main/java/org/odk/collect/android/activities/GoogleSheetsUploaderActivity.java

```diff
@@ -550,18 +558,23 @@ public void authRequest(Uri url, HashMap<String, String> doneSoFar) {
                     selectionArgs[i] = values[i].toString();
                 }
             }
+            String token;
             try {
-                String token = mCredential.getToken();
-
+                token = mCredential.getToken();
                 //Immediately invalidate so we get a differnet one if we have to try again
                 GoogleAuthUtil.invalidateToken(GoogleSheetsUploaderActivity.this, token);
 
+                getIDOfFolderWithName(GOOGLE_DRIVE_ROOT_FOLDER, null);
```

Comment:
I believe this could result in some exceptions being thrown and they should be handled. But I think it makes sense to do all the folder getting and creation together in `uploadOneSubmission`. You don't really gain anything by doing it here.